src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonTypeInfo.cs

```diff
@@ -51,9 +64,18 @@ public partial class JsonTypeInfo
         {
             get
             {
-                if (_elementTypeInfo == null && ElementType != null)
+                if (_elementTypeInfo == null)
                 {
-                    _elementTypeInfo = Options.GetOrAddJsonTypeInfo(ElementType);
+                    if (ElementType != null)
+                    {
+                        // GetOrAddJsonTypeInfo already ensures JsonTypeInfo is configured
+                        // also see comment on JsonPropertyInfo.JsonTypeInfo
+                        _elementTypeInfo = Options.GetOrAddJsonTypeInfo(ElementType);
+                    }
+                }
+                else
+                {
+                    _elementTypeInfo.EnsureConfigured();
```

Comment:
Per your comment in the branch above, why would we need to call `EnsureConfigured` if this has already been populated?

Reply:
as I mentioned in similar place, source gen can directly assign `_elementTypeInfo` in which case it will not be configured.